gitlabform/processors/project/schedules_processor.py

```diff
@@ -29,115 +37,130 @@ def _process_configuration(self, project_and_group: str, configuration: dict):
                 if schedule_ids:
                     debug("Deleting pipeline schedules '%s'", schedule_description)
                     for schedule_id in schedule_ids:
-                        self.gitlab.delete_pipeline_schedule(
-                            project_and_group, schedule_id
-                        )
+                        project.pipelineschedules.get(schedule_id).delete()
                 else:
                     debug(
                         "Not deleting pipeline schedules '%s', because none exist",
                         schedule_description,
                     )
             else:
+                entity_config = configured_schedules[schedule_description]
+
                 if schedule_ids and len(schedule_ids) == 1:
-                    debug(
-                        "Changing existing pipeline schedule '%s'", schedule_description
-                    )
-                    schedule_id = schedule_ids[0]
-                    self.gitlab.take_ownership(project_and_group, schedule_id)
-                    self.gitlab.update_pipeline_schedule(
-                        project_and_group,
-                        schedule_id,
-                        configuration.get("schedules|" + schedule_description),
```

Comment:
Previously updating an existing schedule had involved taking ownership of that schedule first and then performing the update. In the current code, I don't see the "taking ownership" logic anymore. I can't tell what the impact of this is because we didn't have any particular test case around it and I'm not sure what python-gitlab library does when calling `project.pipelineschedules.update` function. I believe taking ownership of schedule needs to be specifically called but I could be wrong. Could you please look into this?

Reply:
@TimKnight-Opencast - latest changes looks good to me. I feel this is the only thread/item outstanding.

I wasn't sure why "take ownership" logic was there. I browsed the older commits and traced the logic being added via [this commit](https://github.com/gitlabform/gitlabform/commit/dd5086b88cdcb642a55585a5ec5d3333eea54998), which seem to have addressed the issue reported in #364 . From one of the comments in that thread, I traced the issue in GitLab's [change in version 14.10.1](https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/changelogs/archive-14.md#14101-2022-04-29).

> Prevent maintainers from editing PipelineSchedule (merge request)

I believe we should re-add this logic back, unless we can confirm that GitLab allows maintainers/other users to be able to edit pipeline schedule.

Reply:
Will add this logic back in, thanks 👍 
Was thinking I was missing comments somewhere

Reply:
Looks like we still need take ownership: https://docs.gitlab.com/ee/ci/pipelines/schedules.html#take-ownership

Reply:
@amimas re-added

Reply:
Thanks for adding the "take ownership" logic back.

I don't think that section of the document actually explains it. The document mentions "how to take ownership" but it doesn't explain why or even the fact that modifying someone else' schedule requires "taking ownership". Or maybe I'm just misunderstanding the doc 😂 .